src/quickcheck.rs

```diff
@@ -0,0 +1,259 @@
+        );
+        Self::from_hms_nanos_unchecked(hour, minute, second, nanosecond)
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        let orig_h = self.hour;
+        let orig_m = self.minute;
+        let orig_s = self.second;
+        let orig_nanos = self.nanosecond;
+
+        let shrunk_hour = orig_h
+            .shrink()
+            .map(move |h| Self::from_hms_nanos_unchecked(h, orig_m, orig_s, orig_nanos));
+        let shrunk_minute = orig_m
+            .shrink()
+            .map(move |m| Self::from_hms_nanos_unchecked(orig_h, m, orig_s, orig_nanos));
+        let shrunk_second = orig_s
+            .shrink()
+            .map(move |s| Self::from_hms_nanos_unchecked(orig_h, orig_m, s, orig_nanos));
+        let shrunk_nanos = orig_nanos
+            .shrink()
+            .map(move |nanos| Self::from_hms_nanos_unchecked(orig_h, orig_m, orig_s, nanos));
+
+        Box::new(
+            shrunk_hour
+                .chain(shrunk_minute)
+                .chain(shrunk_second)
+                .chain(shrunk_nanos),
+        )
+    }
+}
+
+impl Arbitrary for PrimitiveDateTime {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        Self::new(Date::arbitrary(g), Time::arbitrary(g))
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        let orig_date = self.date;
+        let orig_time = self.time;
+
+        let shrunk_date = orig_date.shrink().map(move |d| Self::new(d, orig_time));
+        let shrunk_time = orig_time.shrink().map(move |t| Self::new(orig_date, t));
+
+        Box::new(shrunk_date.chain(shrunk_time))
+    }
+}
+
+impl Arbitrary for UtcOffset {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        let size = min(
+            g.size().try_into().unwrap_or(i32::max_value()),
+            60 * 60 * 24,
+        );
+        let offset = g.gen_range(-(size - 1), size);
+        Self::seconds_unchecked(offset)
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        Box::new(self.as_seconds().shrink().map(Self::seconds_unchecked))
+    }
+}
+
+impl Arbitrary for OffsetDateTime {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        let datetime = PrimitiveDateTime::arbitrary(g);
+        let offset = UtcOffset::arbitrary(g);
+        datetime.assume_offset(offset)
+    }
+
+    fn shrink(&self) -> Box<dyn Iterator<Item = Self>> {
+        let orig_datetime = PrimitiveDateTime::new(self.date(), self.time());
+        let orig_offset = self.offset();
+
+        let shrunk_datetime = orig_datetime
+            .shrink()
+            .map(move |dt| dt.assume_offset(orig_offset));
+        let shrunk_offset = orig_offset
+            .shrink()
+            .map(move |o| orig_datetime.assume_offset(o));
+
+        Box::new(shrunk_datetime.chain(shrunk_offset))
+    }
+}
+
+impl Arbitrary for Weekday {
+    fn arbitrary<G: Gen>(g: &mut G) -> Self {
+        use Weekday::*;
+        match g.gen_range(0, max(1, min(7, g.size()))) {
```

Comment:
```suggestion
        match g.gen_range(0, g.size().clamp(1, 7)) {
```

Reply:
Oh, good catch! Guess I missed that one because it wasn't spanning 5 lines or whatever... :laughing: